devlib/target.py

```diff
@@ -244,8 +244,8 @@ def connect(self, timeout=None, check_boot_completed=True):
         if check_boot_completed:
             self.wait_boot_complete(timeout)
         self._resolve_paths()
-        self.execute('mkdir -p {}'.format(self.working_directory))
-        self.execute('mkdir -p {}'.format(self.executables_directory))
+        self.execute('mkdir -p {0} && chown {1} {0}'.format(self.working_directory, self.user), as_root=True)
```

Comment:
This breaks for unrooted devices. Leave the old method -- it's up to the user to specify a working directory location that is writable by `self.user`.

Reply:
ok